### DISCUSSION TITLE
Object.assign is different with .set() ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [fabricjs/fabric.js#10252] [Q&A]
Checked the source code, .set did nothing else, but Object.assign doesn't work, why is that?
https://github.com/fabricjs/fabric.js/blob/master/src/CommonMethods.ts#L29

```
const rectRef = useRef()
useEffect(()=>{
rectRef.current = new React()
},[])

useEffect(()=>{
Object.assign(rectRef.current,{top:20}) //  Actually this is not work 
rectRef.current.set({top:20})
},[options])

```

**Comment by asturur** [comment]
Set does call specific side effects for specific properties.
Each class has its own side effects.
